### DISCUSSION TITLE
Add mentions, channel_mentions and role_mentions to Embed

### DISCUSSION THREAD

**Discussion by alexisig** [Rapptz/discord.py#7076] [Ideas] [answered]
I'd like to create a PR that adds easy access to mentions inside `Embed` instances (in description, fields etc ...), the same way the `Message` class does (e.g. `message.role_mentions`).

Any thoughts about this?

**Comment by redgoldlace** [comment, marked as answer]
I'm fairly sure this has been discussed in discord.py's discord server previously. It's subject to a couple issues:
- `Message.mentions` (and I believe `Message.role_mentions` too) are provided by the API. The library doesn't handle these itself.
- To my knowledge, the various mentions attributes of `Message` all represent mentions that have *actually* happened - i.e, if you send a message with a mention and set `allowed_mentions` as `AllowedMentions.none()`, I'm fairly sure the corresponding `Message` object will have nothing in its `mentions` attribute. Introducing something like this for embeds when no actual mentions have happened would go against the existing behaviour of the API and serves as an easy pitfall for users. 
- Resolving mentions correctly could be expensive in one way or another, and could easily fall out of sync with the client's behaviour. The client doesn't render mentions in some embed positions (i.e, titles) but this behaviour can change between client versions. Should we mimic the client at all costs, or should we return a user/member if there is an attempted mention in any part of the embed, even if the client will not attempt to resolve it? And etc. It's a lot of maintenance work.
- You have no guarantees if any of the IDs you fish out of the embed are actually valid. This doesn't play nicely if the user has no member cache, either, as checking if the user is valid would likely require an API request, which shouldn't be implicit behaviour. Likewise the user on the client end may be able to "see" members that a bot can't, so even if API requests were made, it could still be ambiguous as to whether it was a valid user or member.
- What would the return type of the function be? An embed's content is just a string, and fishing through that for mentions only gives you raw IDs. Checking the validity of those IDs is fraught with issues, so I feel there isn't really a sensible return type for something like this. A list of `int`s is probably the closest to sensible you could get, but it's still not great.

I'm not sure if an implementation that doesn't suffer from these issues is really possible, which is why it hasn't happened yet.
